zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -308,64 +335,172 @@ benchmarks! {
+            market_id,
+            min_liquidity
+        )
+        .unwrap();
 
         let weight_len: usize = MaxRuntimeUsize::from(a).into();
         let weights = vec![MinWeight::get(); weight_len];
-    }: _(RawOrigin::Signed(caller), market_id, max_swap_fee, min_liquidity, weights)
 
-    dispute {
-        let a in 0..(T::MaxDisputes::get() - 1) as u32;
-        let (caller, market_id) = create_close_and_report_market::<T>(
+        let call = Call::<T>::deploy_swap_pool_for_market {
+            market_id,
+            swap_fee: max_swap_fee,
+            amount: min_liquidity,
+            weights
+        };
+    }: {
+        call.dispatch_bypass_filter(RawOrigin::Signed(caller).into())?;
+    } verify {
+        let current_len = MarketIdsPerOpenTimeFrame::<T>::get(
+            Pallet::<T>::calculate_time_frame_of_moment(range_start),
+        )
+        .len();
+        assert!(current_len == prev_len + 1);
+    }
+
+    deploy_swap_pool_for_market_open_pool {
+        let a in (T::MinCategories::get().into())..T::MaxCategories::get().into();
+
+        // We need to ensure, that period range start is now,
+        // because we would like to open the pool now
+        let range_start: MomentOf<T> = T::MarketCommons::now();
+        let range_end: MomentOf<T> = 1_000_000u64.saturated_into();
+        let (caller, market_id) = create_market_common::<T>(
             MarketCreation::Permissionless,
-            MarketType::Scalar(0u128..=u128::MAX),
-            OutcomeReport::Scalar(42)
+            MarketType::Categorical(a.saturated_into()),
+            ScoringRule::CPMM,
+            Some(MarketPeriod::Timestamp(range_start..range_end)),
         )?;
-    }:  {
-        let origin = caller.clone();
-        let disputes = crate::Disputes::<T>::get(market_id);
-        let market = T::MarketCommons::market(&Default::default()).unwrap();
-        T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?;
+
+        let market = T::MarketCommons::market(&market_id.saturated_into()).unwrap();
+
+        let max_swap_fee: BalanceOf::<T> = MaxSwapFee::get().saturated_into();
+        let min_liquidity: BalanceOf::<T> = MinLiquidity::get().saturated_into();
+        let _ = Pallet::<T>::buy_complete_set(
+            RawOrigin::Signed(caller.clone()).into(),
+            market_id,
+            min_liquidity
+        )
+        .unwrap();
+
+        let weight_len: usize = MaxRuntimeUsize::from(a).into();
+        let weights = vec![MinWeight::get(); weight_len];
+
+        let call = Call::<T>::deploy_swap_pool_for_market {
+            market_id, swap_fee: max_swap_fee, amount: min_liquidity, weights
+        };
+    }: {
+        call.dispatch_bypass_filter(RawOrigin::Signed(caller).into())?;
+    } verify {
+        let market_pool_id = T::MarketCommons::market_pool(&market_id.saturated_into()).unwrap();
+        let pool = T::Swaps::pool(market_pool_id).unwrap();
+        assert!(pool.pool_status == PoolStatus::Active);
```

Comment:
[nitpick] You can use `assert_eq!` here.

Reply:
[e3b1c77](https://github.com/zeitgeistpm/zeitgeist/pull/783/commits/e3b1c772ccd1892818e9284934a5e7b3f646c6b7)